Explicitly set GeckoView dependency to have classifier `@aar`

For technical reasons, Gradle module substitution can't change the "classifier type" of substituted modules.  Also for technical reasons, the "default" type of modules is `@jar`, but the Android-Gradle plugin has internal work-arounds to make it find AARs.  Together, this means that module substitution isn't straightforward: substituting `org.mozilla.geckoview:geckoview-nightly` for `org.mozilla.geckoview:geckoview-default` will find `@jar` type modules, not `@aar`.

It is possible to work around this: consumers can always find some way to make `browser-engine-gecko-*` _not_ include the GeckoView dependency, and then include it (or substitute it) themselves.  But it would be easier to do that -- for example, using the approaches of [Bug 1533465](https://bugzilla.mozilla.org/show_bug.cgi?id=1533465) -- if `browser-engine-gecko-*` set the `@aar` classifier itself.  Concretely, that means using:
```
implementation "org.mozilla.geckoview:geckoview-nightly:$VERSION@aar"
```

I am not aware of any repercussions from setting the classifier explicitly, but it's possible there are some that I'm not aware of.

@ncalexan I still see the snakeyaml issue with GeckoView 70.0.20190818094547 running the sample browser on a Nexus 5 (Android 6.0.1).

```
                         E  FATAL EXCEPTION: main
                         E  Process: org.mozilla.samples.browser, PID: 18078
                         E  java.lang.NoClassDefFoundError: Failed resolution of: Lorg/yaml/snakeyaml/constructor/Constructor;
                         E      at org.mozilla.gecko.util.DebugConfig.fromFile(DebugConfig.java:40)
                         E      at org.mozilla.geckoview.GeckoRuntime.init(GeckoRuntime.java:279)
                         E      at org.mozilla.geckoview.GeckoRuntime.getDefault(GeckoRuntime.java:153)
                         E      at mozilla.components.browser.engine.gecko.GeckoEngine.<init>(GeckoEngine.kt:37)
                         E      at org.mozilla.samples.browser.Components$engine$2.invoke(Components.kt:16)
                         E      at org.mozilla.samples.browser.Components$engine$2.invoke(Components.kt:14)
                         E      at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
                         E      at org.mozilla.samples.browser.Components.getEngine(Components.kt)
                         E      at org.mozilla.samples.browser.SampleApplication.onCreate(SampleApplication.kt:28)
                         E      at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1013)
                         E      at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4707)
                         E      at android.app.ActivityThread.-wrap1(ActivityThread.java)
                         E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1405)
                         E      at android.os.Handler.dispatchMessage(Handler.java:102)
                         E      at android.os.Looper.loop(Looper.java:148)
                         E      at android.app.ActivityThread.main(ActivityThread.java:5417)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                         E  Caused by: java.lang.ClassNotFoundException: Didn't find class "org.yaml.snakeyaml.constructor.Constructor" on path: DexPathList
                            [[zip file "/data/app/org.mozilla.samples.browser-1/base.apk"],nativeLibraryDirectories=[/data/app/org.mozilla.samples.browser-1
                            /lib/arm, /data/app/org.mozilla.samples.browser-1/base.apk!/lib/armeabi-v7a, /vendor/lib, /system/lib]]
                         E      at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
                         E      at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
                         E      at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
                         E      ... 19 more
                         E      Suppressed: java.lang.ClassNotFoundException: org.yaml.snakeyaml.constructor.Constructor
                         E          at java.lang.Class.classForName(Native Method)
                         E          at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
                         E          at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
                         E          at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
                         E          ... 20 more
                         E      Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
```

--- a/samples/browser/build.gradle
+++ b/samples/browser/build.gradle
@@ -182,3 +182,11 @@ dependencies {
     androidTestImplementation Dependencies.androidx_espresso_core
     androidTestImplementation Dependencies.testing_mockwebserver
 }
+
+if (gradle.hasProperty('localProperties.dependencySubstitutions.geckoviewTopsrcdir')) {
+    if (gradle.hasProperty('localProperties.dependencySubstitutions.geckoviewTopobjdir')) {
+        ext.topobjdir = gradle."localProperties.dependencySubstitutions.geckoviewTopobjdir"
+    }
+    ext.topsrcdir = gradle."localProperties.dependencySubstitutions.geckoviewTopsrcdir"
+    apply from: "${topsrcdir}/substitute-local-geckoview.gradle"
+}

--- a/settings.gradle
+++ b/settings.gradle
@@ -58,6 +58,10 @@ if (file('local.properties').canRead()) {
 }
 
 if (localProperties != null) {
+    localProperties.each { prop ->
+        gradle.ext.set("localProperties.${prop.key}", prop.value)
+    }
+
     String appServicesLocalPath = localProperties.getProperty(settingAppServicesPath);
 
     if (appServicesLocalPath != null) {

